Source/dvlnet/tcp_client.cpp

```diff
@@ -28,10 +29,40 @@ int tcp_client::join(std::string addrstr)
 	constexpr int MsSleep = 10;
 	constexpr int NoSleep = 250;
 
-	std::string port = StrCat(*sgOptions.Network.port);
+	const char *defaultPort = "6112";
+	std::string_view host = "";
+	std::string_view port = defaultPort;
+	if (addrstr[0] == '[') {
+		// Assume IPv6 address in square brackets, followed by port
+		// Example: [::1]:6113
+		size_t pos = addrstr.find(']', 1);
+		pos = pos != std::string::npos ? pos + 1 : addrstr.length();
+		host = std::string_view(addrstr.c_str(), pos);

```

Comment:
If the argument is changed to `std::string_view`, this can be `host = addrstr.substr(0, pos)`